engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -429,6 +432,15 @@ def handle_response(response)
             raise ActionController::TranslatedError.new(error['error'])
           end
         end
+
+        def no_more_activations(scc_systems_activations)
```

Comment:
maybe rename to `no_more_activations?`

Reply:
We end up having to to a negative test anyway either via name, as in `no_more_activations` or via a `not` condition in the test. As such I think it makes sense to name the method according to what we are actually looking for which is non-free extension. I'd suggest `product_has_paid_extension` or `product_has_paid_extension_active` or `system_has_paid_extension`, something along those lines.

Reply:
OK, `system_has_paid_extension` as it's the thing object with extensions (products)